Geometry/EcalCommonData/src/EcalBaseNumber.cc

```diff
@@ -25,7 +25,7 @@ int EcalBaseNumber::getCopyNumber(int level) const { return _sortedCopyNumber[le
 
 int EcalBaseNumber::getCopyNumber(const std::string& levelName) const {
   for (int iLevel = 0; iLevel < _theLevels; iLevel++) {
-    if (_sortedName[iLevel] == levelName) {
+    if (_sortedName[iLevel].find(levelName) != std::string::npos) {
```

Comment:
@bsunanda If a volume of a given hierarchy level has a name partly containing the name of another hierarchy level volume, this will lead to issues.

The underlying, dd4hep-migration related issue, is that the G4 logical volumes, as produced from DD4hep CMSSW interface, have different names, than the G4 logical volumes names as produced from old DD.

I think the cleanest (and fastest) approach would be to have a simple string manipulation **upstream** to get the same G4 logical volumes with DD4hep. Ie, casting namespace and removing the extra _copyNumber from the logical volume name.

I have a branch where it is done.
This solves this specific issue, but also all the other names-related ones. Otherwise one has to change the string manipulations in all the client code, in all the string-related functions.